app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt

```diff
@@ -115,6 +126,45 @@ class FinishedFlightsViewModel(val repository: Repository, val userId: String) :
             }
       }
 
+  private fun searchLocation(query: String) =
+      viewModelScope.launch(Dispatchers.IO) {
+        val client = OkHttpClient()
+        val url = "https://nominatim.openstreetmap.org/search?q=$query&format=json&limit=4"
+        val request = Request.Builder().url(url).build()
+        _searchResponse.value =
+            try {
+              val response = client.newCall(request).execute()
+              val responseBody = response.body()?.string()
+              responseBody
+            } catch (e: Exception) {
+              e.printStackTrace()
+              "error"
+            }
+      }
+
+  fun getSearchLocation(query: String, time: Int) =
+      viewModelScope.launch {
+        searchLocation(query).join()
+        val gson = Gson()
+        if (_searchResponse.value == null) {
+          _searchResults.value = null
+        } else if (_searchResponse.value != null && _searchResponse.value != "") {
+          Log.d("searchLocation", _searchResponse.value!!)
+          val jsonArray = gson.fromJson(_searchResponse.value, JsonArray::class.java)
+          Log.d("searchLocation", jsonArray.toString())
```

Comment:
Looks like debugging leftovers, you can remove them

Reply:
Indead, it is now removed